llm: add page (#11697)
tldr-pages/tldr: pages/common/llm.md

> # LLM

Reviewer: ```suggestion
# llm
```
Note. The page headings must match command name.

> > A utility and Python library for interacting with Large Language Models (LLMs) via remote APIs and models that can be installed and run on your machine.

Reviewer 2: ```suggestion
> Interact with Large Language Models (LLMs) via remote APIs and models that can be installed and run on your machine.
```

> More information: <https://llm.datasette.io/en/stable/help.html>.

> - Set up OpenAI API Key:

Reviewer 2: ```suggestion
- Set up an OpenAI API Key:
```

`llm keys set openai`

- Run a prompt:

`llm "{{Ten fun names for a pet pelican}}"`

- Run a [s]ystem prompt against a file:

> `cat {{path/to/file.py}} | llm -s "{{Explain this code}}"`

Reviewer 2: Please use long options if available: https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#option-syntax
```suggestion
`cat {{path/to/file.py}} | llm --system "{{Explain this code}}"`
```

- Install packages from PyPI into the same environment as LLM:

`llm install {{package1 package2 ...}}`

- Download and run a prompt against a [m]odel:

> `llm -m {{orca-mini-3b-gguf2-q4_0}} '{{What is the capital of France?}}'`

Reviewer 2: ```suggestion
`llm --model {{orca-mini-3b-gguf2-q4_0}} "{{What is the capital of France?}}"`
```

- Have an interactive chat with a specific [m]odel:

> `llm chat -m {{chatgpt}}`

Reviewer 2: ```suggestion
`llm chat --model {{chatgpt}}`
```

# LLM

> A CLI utility and Python library for interacting with Large Language Models, both via remote APIs and models that can be installed and run on your own machine.
> More information: <https://llm.datasette.io/en/stable/>.

- Install LLM using pip:

`pip install llm`

- Install LLM using pipx:

`pipx install llm`

- Set up OpenAI API Key:

`llm keys set openai`

- Run a Prompt:

`llm "Ten fun names for a pet pelican"`

- Run a system prompt against a file:

`cat myfile.py | llm -s "Explain this code"`

> - Install Plugin and Local Models:
> 
> `llm install llm-gpt4all`

Reviewer: ```suggestion
- Install packages from PyPI into the same environment as LLM:

`llm install {{package1 package 2 ...}}`
```

> - Download and run a prompt against the Orca Mini 7B model:
> 
> `llm -m orca-mini-3b-gguf2-q4_0 'What is the capital of France?'`

Reviewer: ```suggestion
- Download and run a prompt against a [m]odel:

`llm -m {{orca-mini-3b-gguf2-q4_0}} '{{What is the capital of France?}}'`
```

> - Interactive Chat:
> 
> `llm chat -m chatgpt`

Reviewer: ```suggestion
- Have an interactive chat with a specific [m]odel:

`llm chat -m {{chatgpt}}`
```
